client/settings/fraud-protection/advanced-settings/utils.ts

```diff
@@ -96,117 +98,83 @@ const buildRuleset = (
 			};
 			break;
 		case Rules.RULE_ORDER_ITEMS_THRESHOLD:
-			if (
-				parseInt(
-					ruleConfiguration?.min_items?.toString() || '',
-					10
-				) &&
-				parseInt( ruleConfiguration?.max_items?.toString() || '', 10 )
-			) {
-				ruleBase.check = {
-					operator: CheckOperators.LIST_OPERATOR_OR,
-					checks: [
-						{
-							key: Checks.CHECK_ITEM_COUNT,
-							operator: CheckOperators.OPERATOR_LT,
-							value:
-								parseInt(
-									ruleConfiguration?.min_items?.toString() ||
-										'',
-									10
-								) ?? null,
-						},
-						{
-							key: Checks.CHECK_ITEM_COUNT,
-							operator: CheckOperators.OPERATOR_GT,
-							value:
-								parseInt(
-									ruleConfiguration?.max_items?.toString() ||
-										'',
-									10
-								) ?? null,
-						},
-					],
-				};
-			} else if (
-				parseInt(
-					ruleConfiguration?.min_items?.toString() || '',
-					10
-				) ||
-				parseInt( ruleConfiguration?.max_items?.toString() || '', 10 )
-			) {
-				ruleBase.check = parseInt(
-					ruleConfiguration?.min_items?.toString() || '',
-					10
-				)
-					? {
-							key: Checks.CHECK_ITEM_COUNT,
-							operator: CheckOperators.OPERATOR_LT,
-							value:
-								parseInt(
-									ruleConfiguration?.min_items?.toString() ||
-										'',
-									10
-								) ?? null,
-					  }
-					: {
-							key: Checks.CHECK_ITEM_COUNT,
-							operator: CheckOperators.OPERATOR_GT,
-							value:
-								parseInt(
-									ruleConfiguration?.max_items?.toString() ||
-										'',
-									10
-								) ?? null,
-					  };
+			if ( isOrderItemsThresholdSetting( ruleConfiguration ) ) {
```

Comment:
Why didn't we directly use 

```ts
if ( ruleConfiguration typeof FraudPreventionOrderItemsThresholdSetting ) {
```

here? I think a single check like this will be enough than creating separate functions to check the type.

Reply:
Thanks for the suggestion here. I gave it a try but both approaches result in errors.

```typescript
if ( ruleConfiguration instanceof FraudPreventionOrderItemsThresholdSetting ) {
```

> 'FraudPreventionOrderItemsThresholdSetting' only refers to a type, but is being used as a value here.

```typescript
if ( typeof ruleConfiguration === 'FraudPreventionOrderItemsThresholdSetting' ) {
```
> This comparison appears to be unintentional because the types '"string" | "number" | "bigint" | "boolean" | "symbol" | "undefined" | "object" | "function"' and '"FraudPreventionOrderItemsThresholdSetting"' have no overlap.